src/ImageSharp/Memory/DiscontiguousBuffers/MemoryGroupExtensions.cs

```diff
@@ -61,6 +61,35 @@ internal static Memory<T> GetBoundedSlice<T>(this IMemoryGroup<T> group, long st
             return memory.Slice(bufferStart, length);
         }
 
+        /// <summary>
+        /// Returns the slice of the buffer starting at global index <paramref name="start"/> that goes until the end of the buffer.
+        /// </summary>
+        internal static Memory<T> GetRemainingSliceOfBuffer<T>(this IMemoryGroup<T> group, long start)
+            where T : struct
+        {
+            Guard.NotNull(group, nameof(group));
+            Guard.IsTrue(group.IsValid, nameof(group), "Group must be valid!");
+            Guard.MustBeLessThan(start, group.TotalLength, nameof(start));
+
+            int bufferIdx = (int)(start / group.BufferLength);
+
+            if (bufferIdx < 0)
```

Comment:
Can combine with the check below by using
```c#
if ((uint)bufferIdx >= group.Count)
{
   ...
}
```
A negative value for `bufferIdx` is after the `uint` cast in the range > int.MaxValue, so definitely >= group.Count too.